### PR TITLE
Add React and support primitive goal setting in the 3D scene

### PR DESCRIPTION
**Pull request by isherman**

### CODE REVIEW

File: app/frontend/package.json [hunk 1]
```diff
@@ -12,19 +12,24 @@
...
+    "eslint-config-airbnb": "18.1.0",
     "eslint-config-prettier": "^6.11.0",
-    "eslint-plugin-import": "^2.20.2",
+    "eslint-plugin-import": "^2.20.1",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-prettier": "^3.1.3",
-    "eslint-plugin-react": "^7.20.0",
+    "eslint-plugin-react": "^7.19.0",
+    "eslint-plugin-react-hooks": "2.5.0",
     "prettier": "^2.0.5",
```
Comment (ethanrublee) [inline app/frontend/package.json:23]: Here i get this error:

```
rublee@farmws01:~/code/tractor/app/frontend$ yarn
yarn install v1.22.4
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
error prettier@2.0.5: The engine "node" is incompatible with this module. Expected version ">=10.13.0". Got "8.10.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
rublee@farmws01:~/code/tractor/app/frontend$ which node
/usr/bin/node
rublee@farmws01:~/code/tractor/app/frontend$ node --version
v8.10.0
```

Comment (isherman) [reply app/frontend/package.json:23]: Let me know if you still have a problem with your updated node version.

**Comment by ethanrublee** [comment]
I find if i orbit and then release it adds a landmark.  Tried uploading a gif but having trouble getting it to work.

**Comment by isherman** [comment]
> I find if i orbit and then release it adds a landmark. Tried uploading a gif but having trouble getting it to work.

I believe that! Good QA :) Happy to fix.

**Comment by isherman** [comment]
> > I find if i orbit and then release it adds a landmark. Tried uploading a gif but having trouble getting it to work.
> 
> I believe that! Good QA :) Happy to fix.

OK, now waypoints are only created when *shift* is depressed. Waypoints can be set as goals with just a regular click. Orbit should work as expected.

**Comment by isherman** [comment]
I guess that conflicts with Three.js's Dolly Camera. Open to ideas.

**Comment by ethanrublee** [comment]
Looks like orbit is somehow a drag event (haven't looked too closely at the code). When I left drag and release it places a marker where the mouse releases.  Maybe a double click? Or if there is the difference between a drag and short click event.  Haven't tested on a touch screen.  Some way to toggle goal placement versus camera motion...

**Comment by isherman** [comment]
> Looks like orbit is somehow a drag event (haven't looked too closely at the code). When I left drag and release it places a marker where the mouse releases. Maybe a double click? Or if there is the difference between a drag and short click event. Haven't tested on a touch screen. Some way to toggle goal placement versus camera motion...

Changed to the alt key for now, and added instruction at the top of the page. Don't want to spend too much optimizing for mobile, etc.

**Pull request closed without merging**